src/Search.php

```diff
@@ -283,60 +283,64 @@ public function load(
 	/**
 	 * Gets the paged posts that belong to the next page/load and are to be used with the twig template.
 	 */
-	public static function get_paged_posts() {
-		// If this is an ajax call.
-		if ( wp_doing_ajax() ) {
-			$search_action = filter_input( INPUT_GET, 'search-action', FILTER_SANITIZE_STRING );
-			$paged         = filter_input( INPUT_GET, 'paged', FILTER_SANITIZE_STRING );
-
-			// Check if call action is correct.
-			if ( 'get_paged_posts' === $search_action ) {
-				$search_async = new static();
-				$search_async->set_context( $search_async->context );
-				$search_async->search_query = urldecode( filter_input( INPUT_GET, 'search_query', FILTER_SANITIZE_STRING ) );
-
-				// Get the decoded url query string and then use it as key for redis.
-				$query_string_full = urldecode( filter_input( INPUT_SERVER, 'QUERY_STRING', FILTER_SANITIZE_STRING ) );
-				$query_string      = str_replace( '&query-string=', '', strstr( $query_string_full, '&query-string=' ) );
-
-				$search_async->current_page = $paged;
-
-				parse_str( $query_string, $filters_array );
-				$selected_sort    = $filters_array['orderby'] ?? self::DEFAULT_SORT;
-				$selected_filters = $filters_array['f'] ?? [];
-				$filters          = [];
-
-				// Handle submitted filter options.
-				if ( $selected_filters && is_array( $selected_filters ) ) {
-					foreach ( $selected_filters as $type => $filter_type ) {
-						if ( ! is_array( $filter_type ) ) {
-							continue;
-						}
-						foreach ( $filter_type as $name => $id ) {
-							$filters[ $type ][] = [
-								'id'   => $id,
-								'name' => $name,
-							];
-						}
-					}
-				}
+	public static function get_paged_posts(): void {
+		if ( ! wp_doing_ajax() ) {
```

Comment:
Also here not sure if it even makes sense to do this check. If a function calls `get_paged_posts` from a non-ajax context, it's doing the wrong thing and won't get the intended effect anyway.